src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,212 @@
+#:include "common.fypp"
+module stdlib_stats_distribution_PRNG
+    use stdlib_kinds, only: int8, int16, int32, int64
+    implicit none
+    private
+    integer, parameter :: MAX_INT_BIT_SIZE = bit_size(1_int64)
+    integer(int64), save :: st(4), si = 614872703977525537_int64
+    logical, save :: seed_initialized = .false.
+
+    public :: random_seed
+    public :: dist_rand
+    public :: jump
+    public :: long_jump
+
+
+    interface dist_rand
+    !! Version experimental
+    !!
+    !! Generation of random integers with different kinds
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure dist_rand_${t1[0]}$${k1}$
+        #:endfor
+    end interface dist_rand
+
+    interface random_seed
+    !! Version experimental
+    !!
+    !! Set seed value for random number generator
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+    !!
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure random_distribution_seed_${t1[0]}$${k1}$
+        #:endfor
+    end interface random_seed
+
+
+    contains
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    function dist_rand_${t1[0]}$${k1}$(n) result(res)
+    !! Random integer generation for various kinds
+    !! result = [-2^k, 2^k - 1], k = 7, 15, 31, 63, depending on input kind
+    !! Result will be operated by bitwise operators to generate desired integer
+    !! and real pseudorandom numbers
+    !!
+        ${t1}$, intent(in) :: n
+        ${t1}$ :: res
+        integer :: k
+
+        k = MAX_INT_BIT_SIZE - bit_size(n)
```

Comment:
If the `INTS_KINDS_TYPES` fypp macro is ever extended to include 128-bit integers (available with gfortran), then I think this line will break. I don't think we need to act upon this now, just as a thought.

Reply:
Since Fortran standard does not include int128 integer, the MAX_INT_BIT_SIZES is set 64. I can insert a condition to limit integer to 64 bit.